src/z_data/serde/de.rs

```diff
@@ -116,7 +116,8 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
     {
         match self.input {
             ZData::Atom(x) if x.len() == 2 => {
-                let a: [u8; 2] = x.clone().try_into().unwrap();
+                let x_clone = x.clone();
+                let a: [u8; 2] = [x_clone[0], x_clone[1]];
```

Comment:
This change doesn't modify how the code works: if `x_clone[0]` or `x_clone[1]` is out of bounds, there will be a runtime panic.
An alternative to prevent the panic is to add a `match` block to handle the result of `x.clone().try_into()`, and return an `Err` if it is not `Ok`. (Similar to the code inside `deserialize_u32`)